pkg/arvo/app/invite-store.hoon

```diff
@@ -0,0 +1,186 @@
+/+  *invite-json
+|%
++$  move  [bone card]
+::
++$  card
+  $%  [%diff diff]
+      [%quit ~]
+  ==
+::
++$  state
+  $%  [%0 state-zero]
+  ==
+::
++$  state-zero
+  $:  =invites
+  ==
+::
++$  diff
+  $%  [%invite-initial invites]
+      [%invite-update invite-update]
+  ==
+--
+::
+|_  [bol=bowl:gall state]
+::
+++  this  .
+::
+++  prep
+  |=  old=(unit state)
+  ^-  (quip move _this)
+  ?~  old
+    [~ this]
+  [~ this(+<+ u.old)]
+::
+++  peek-x-all
+  |=  pax=path
+  ^-  (unit (unit [%noun (map path invitatory)]))
+  [~ ~ %noun invites]
+::
+++  peek-x-invitatory
+  |=  pax=path
+  ^-  (unit (unit [%noun (unit invitatory)]))
+  ?~  pax
+    ~
+  =/  invitatory=(unit invitatory)  (~(get by invites) pax)
+  [~ ~ %noun invitatory]
+::
+++  peek-x-invite
+  |=  pax=path
+  ^-  (unit (unit [%noun (unit invite)]))
+  ?~  pax
+    ~
+  =/  invitatory=(unit invitatory)  (~(get by invites) t.pax)
+  ?~  invitatory
+    ~
+  =/  uid=serial  (slav %uv i.pax)
+  =/  invite=(unit invite)  (~(get by u.invitatory) uid)
+  [~ ~ %noun invite]
+::
+++  peer-all
+  |=  pax=path
+  ^-  (quip move _this)
+  ?>  (team:title our.bol src.bol)
+  ::  send all updates from now on
+  :_  this
+  [ost.bol %diff %invite-initial invites]~
+::
+++  peer-updates
+  |=  pax=path
+  ^-  (quip move _this)
+  ?>  (team:title our.bol src.bol)
+  ::  send all updates from now on
+  [~ this]
+::
+++  peer-invitatory
+  |=  pax=path
+  ^-  (quip move _this)
+  ?>  (team:title our.bol src.bol)
+  =/  inv=(unit invitatory)  (~(get by invites) pax)
+  ?~  inv  !!
+  :_  this
+  [ost.bol %diff %invite-update [%invitatory u.inv]]~
+::
+++  poke-json
+  |=  =json
+  ^-  (quip move _this)
+  ?>  (team:title our.bol src.bol)
+  (poke-invite-action (json-to-action json))
+::
+++  poke-invite-action
+  |=  action=invite-action
+  ^-  (quip move _this)
+  ?>  (team:title our.bol src.bol)
+  ?-  -.action
+      %create
+    (handle-create action)
```

Comment:
Probably just one-line all of these, ie `%create  (handle-create action`.

Reply:
Good idea.